tests/integration/cloudformation/test_cloudformation_changesets.py

```diff
@@ -0,0 +1,344 @@
+
+def test_create_change_set_invalid_params(cfn_client):
+    stack_name = f"stack-{short_uid()}"
+    change_set_name = f"change-set-{short_uid()}"
+    with pytest.raises(ClientError) as ex:
+        cfn_client.create_change_set(
+            StackName=stack_name,
+            ChangeSetName=change_set_name,
+            TemplateBody=load_template_raw("sns_topic_simple.yaml"),
+            ChangeSetType="INVALID",
+        )
+    err = ex.value.response["Error"]
+    assert err["Code"] == "ValidationError"
+
+
+def test_create_change_set_missing_stackname(cfn_client):
+    """in this case boto doesn't even let us send the request"""
+    change_set_name = f"change-set-{short_uid()}"
+    with pytest.raises(Exception):
+        cfn_client.create_change_set(
+            StackName="",
+            ChangeSetName=change_set_name,
+            TemplateBody=load_template_raw("sns_topic_simple.yaml"),
+            ChangeSetType="CREATE",
+        )
+
+
+def test_create_change_set_with_ssm_parameter(
+    cfn_client,
+    sns_client,
+    ssm_client,
+    cleanup_changesets,
+    cleanup_stacks,
+    is_change_set_created_and_available,
+    is_stack_created,
+):
+    """References a simple stack parameter"""
+
+    stack_name = f"stack-{short_uid()}"
+    change_set_name = f"change-set-{short_uid()}"
+    parameter_name = f"ls-param-{short_uid()}"
+    parameter_value = f"ls-param-value-{short_uid()}"
+    sns_topic_logical_id = "topic123"
+    parameter_logical_id = "parameter123"
+
+    ssm_client.put_parameter(Name=parameter_name, Value=parameter_value, Type="String")
+    template = load_template_raw("dynamicparameter_ssm_string.yaml")
+    template_rendered = jinja2.Template(template).render(parameter_name=parameter_name)
+    response = cfn_client.create_change_set(
+        StackName=stack_name,
+        ChangeSetName=change_set_name,
+        TemplateBody=template_rendered,
+        ChangeSetType="CREATE",
+    )
+    change_set_id = response["Id"]
+    stack_id = response["StackId"]
+    assert change_set_id
+    assert stack_id
+
+    try:
+        # make sure the change set wasn't executed (which would create a new topic)
+        list_topics_response = sns_client.list_topics()
+        matching_topics = [
+            t for t in list_topics_response["Topics"] if parameter_value in t["TopicArn"]
+        ]
+        assert matching_topics == []
+
+        # stack is initially in REVIEW_IN_PROGRESS state. only after executing the change_set will it change its status
+        stack_response = cfn_client.describe_stacks(StackName=stack_id)
+        assert stack_response["Stacks"][0]["StackStatus"] == "REVIEW_IN_PROGRESS"
+
+        # Change set can now either be already created/available or it is pending/unavailable
+        wait_until(is_change_set_created_and_available(change_set_id))
+        describe_response = cfn_client.describe_change_set(ChangeSetName=change_set_id)
+
+        assert describe_response["ChangeSetName"] == change_set_name
```

Comment:
nit: Duplication with lines 50-61 - could be a candidate for pulling out into a small util function...

Reply:
I agree, but for now I'd prefer to leave it as is and re-assess it when I add more CloudFormation integration tests if thats fine for you :+1: 
I've now added a corresponding TODO though